### PR TITLE
fix #3881 feat(nimbus): admin workflow for launching experiments

### LINKED ISSUE #3881: Django Admin workflow for Nimbus Experiments
To support an interim workflow of launching Nimbus experiments through the Experimenter Django Admin, let's make sure all the fields that are necessary to fully define a Nimbus experiment are exposed and arranged in the NimbusExperiment admin form.  Additionally we could do the tiniest bit of validation to make sure that branch feature configs validate against the feature config schema.  We should make sure that setting the status to review in the admin causes the celery tasks to pick up and push to remote settings.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-542)
┆Issue Number: EXP-542

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* We'd like to use the Django admin to launch Nimbus experiments before the full UI workflow is ready

This commit

* Rearranges the Nimbus Experiment admin and adds some additional validation
* Updates some targeting constants to more closely match real targeting for about:welcome
* Caught a bug with the nimbus tasks not filtering by collection/application correctly

I tried porting [this about:welcome recipe](https://gist.github.com/punamdahiya/08d139e51f00cc61fd7b6fe30efad535#file-msw-setdefault-v1-recipe-L4) and it came out like this:

```json
{
	"schemaVersion": "1.0.0",
	"slug": "multi-stage-aboutwelcome-set-default-release-82-83",
	"id": "multi-stage-aboutwelcome-set-default-release-82-83",
	"application": "firefox-desktop",
	"userFacingName": "Multi-Stage About:Welcome Set Default v1",
	"userFacingDescription": "Study to validate set default inside onboarding experience",
	"isEnrollmentPaused": false,
	"bucketConfig": {
		"randomizationUnit": "normandy_id",
		"namespace": "multi-stage-aboutwelcome-set-default-release-82-83-1",
		"start": 0,
		"count": 2000,
		"total": 10000
	},
	"probeSets": [],
	"branches": [{
		"slug": "treatment-c",
		"ratio": 1,
		"feature": {
			"featureId": "aboutwelcome",
			"enabled": true,
			"value": {
				"id": "msw-late-default",
				"template": "multistage",
				"screens": [{
					"id": "AW_GET_STARTED",
					"order": 0,
					"content": {
						"zap": true,
						"title": {
							"string_id": "onboarding-multistage-welcome-header"
						},
						"subtitle": {
							"string_id": "onboarding-multistage-welcome-subtitle"
						},
						"primary_button": {
							"label": {
								"string_id": "onboarding-multistage-welcome-primary-button-label"
							},
							"action": {
								"navigate": true
							}
						},
						"secondary_button": {
							"text": {
								"string_id": "onboarding-multistage-welcome-secondary-button-text"
							},
							"label": {
								"string_id": "onboarding-multistage-welcome-secondary-button-label"
							},
							"position": "top",
							"action": {
								"type": "SHOW_FIREFOX_ACCOUNTS",
								"addFlowParams": true,
								"data": {
									"entrypoint": "activity-stream-firstrun"
								}
							}
						}
					}
				}, {
					"id": "AW_IMPORT_SETTINGS",
					"order": 1,
					"content": {
						"zap": true,
						"disclaimer": {
							"string_id": "onboarding-import-sites-disclaimer"
						},
						"title": {
							"string_id": "onboarding-multistage-import-header"
						},
						"subtitle": {
							"string_id": "onboarding-multistage-import-subtitle"
						},
						"tiles": {
							"type": "topsites",
							"showTitles": true
						},
						"primary_button": {
							"label": {
								"string_id": "onboarding-multistage-import-primary-button-label"
							},
							"action": {
								"type": "SHOW_MIGRATION_WIZARD",
								"navigate": true
							}
						},
						"secondary_button": {
							"label": {
								"string_id": "onboarding-multistage-import-secondary-button-label"
							},
							"action": {
								"navigate": true
							}
						}
					}
				}, {
					"id": "AW_CHOOSE_THEME",
					"order": 2,
					"content": {
						"zap": true,
						"title": {
							"string_id": "onboarding-multistage-theme-header"
						},
						"subtitle": {
							"string_id": "onboarding-multistage-theme-subtitle"
						},
						"tiles": {
							"type": "theme",
							"action": {
								"theme": "<event>"
							},
							"data": [{
								"theme": "automatic",
								"label": {
									"string_id": "onboarding-multistage-theme-label-automatic"
								},
								"tooltip": {
									"string_id": "onboarding-multistage-theme-tooltip-automatic-2"
								},
								"description": {
									"string_id": "onboarding-multistage-theme-description-automatic-2"
								}
							}, {
								"theme": "light",
								"label": {
									"string_id": "onboarding-multistage-theme-label-light"
								},
								"tooltip": {
									"string_id": "onboarding-multistage-theme-tooltip-light-2"
								},
								"description": {
									"string_id": "onboarding-multistage-theme-description-light"
								}
							}, {
								"theme": "dark",
								"label": {
									"string_id": "onboarding-multistage-theme-label-dark"
								},
								"tooltip": {
									"string_id": "onboarding-multistage-theme-tooltip-dark-2"
								},
								"description": {
									"string_id": "onboarding-multistage-theme-description-dark"
								}
							}, {
								"theme": "alpenglow",
								"label": {
									"string_id": "onboarding-multistage-theme-label-alpenglow"
								},
								"tooltip": {
									"string_id": "onboarding-multistage-theme-tooltip-alpenglow-2"
								},
								"description": {
									"string_id": "onboarding-multistage-theme-description-alpenglow"
								}
							}]
						},
						"primary_button": {
							"label": {
								"string_id": "onboarding-multistage-theme-primary-button-label"
							},
							"action": {
								"navigate": true
							}
						},
						"secondary_button": {
							"label": {
								"string_id": "onboarding-multistage-theme-secondary-button-label"
							},
							"action": {
								"theme": "automatic",
								"navigate": true
							}
						}
					}
				}, {
					"id": "AW_SET_DEFAULT",
					"order": 3,
					"content": {
						"zap": true,
						"title": "Make Firefox your default browser",
						"subtitle": "Speed, safety, and privacy every time you browse.",
						"primary_button": {
							"label": "Make Default",
							"action": {
								"navigate": true,
								"type": "OPEN_ABOUT_PAGE",
								"data": {
									"args": "preferences",
									"where": "window"
								}
							}
						},
						"secondary_button": {
							"label": {
								"string_id": "onboarding-multistage-import-secondary-button-label"
							},
							"action": {
								"navigate": true
							}
						}
					}
				}]
			}
		}
	}, {
		"slug": "treatment-b",
		"ratio": 1,
		"feature": {
			"featureId": "aboutwelcome",
			"enabled": true,
			"value": {
				"id": "msw-early-default",
				"template": "multistage",
				"screens": [{
					"id": "AW_GET_STARTED",
					"order": 0,
					"content": {
						"zap": true,
						"title": {
							"string_id": "onboarding-multistage-welcome-header"
						},
						"subtitle": {
							"string_id": "onboarding-multistage-welcome-subtitle"
						},
						"primary_button": {
							"label": {
								"string_id": "onboarding-multistage-welcome-primary-button-label"
							},
							"action": {
								"navigate": true
							}
						},
						"secondary_button": {
							"text": {
								"string_id": "onboarding-multistage-welcome-secondary-button-text"
							},
							"label": {
								"string_id": "onboarding-multistage-welcome-secondary-button-label"
							},
							"position": "top",
							"action": {
								"type": "SHOW_FIREFOX_ACCOUNTS",
								"addFlowParams": true,
								"data": {
									"entrypoint": "activity-stream-firstrun"
								}
							}
						}
					}
				}, {
					"id": "AW_SET_DEFAULT",
					"order": 1,
					"content": {
						"zap": true,
						"title": "Make Firefox your default browser",
						"subtitle": "Speed, safety, and privacy every time you browse.",
						"primary_button": {
							"label": "Make Default",
							"action": {
								"navigate": true,
								"type": "OPEN_ABOUT_PAGE",
								"data": {
									"args": "preferences",
									"where": "window"
								}
							}
						},
						"secondary_button": {
							"label": {
								"string_id": "onboarding-multistage-import-secondary-button-label"
							},
							"action": {
								"navigate": true
							}
						}
					}
				}, {
					"id": "AW_IMPORT_SETTINGS",
					"order": 2,
					"content": {
						"zap": true,
						"disclaimer": {
							"string_id": "onboarding-import-sites-disclaimer"
						},
						"title": {
							"string_id": "onboarding-multistage-import-header"
						},
						"subtitle": {
							"string_id": "onboarding-multistage-import-subtitle"
						},
						"tiles": {
							"type": "topsites",
							"showTitles": true
						},
						"primary_button": {
							"label": {
								"string_id": "onboarding-multistage-import-primary-button-label"
							},
							"action": {
								"type": "SHOW_MIGRATION_WIZARD",
								"navigate": true
							}
						},
						"secondary_button": {
							"label": {
								"string_id": "onboarding-multistage-import-secondary-button-label"
							},
							"action": {
								"navigate": true
							}
						}
					}
				}, {
					"id": "AW_CHOOSE_THEME",
					"order": 3,
					"content": {
						"zap": true,
						"title": {
							"string_id": "onboarding-multistage-theme-header"
						},
						"subtitle": {
							"string_id": "onboarding-multistage-theme-subtitle"
						},
						"tiles": {
							"type": "theme",
							"action": {
								"theme": "<event>"
							},
							"data": [{
								"theme": "automatic",
								"label": {
									"string_id": "onboarding-multistage-theme-label-automatic"
								},
								"tooltip": {
									"string_id": "onboarding-multistage-theme-tooltip-automatic-2"
								},
								"description": {
									"string_id": "onboarding-multistage-theme-description-automatic-2"
								}
							}, {
								"theme": "light",
								"label": {
									"string_id": "onboarding-multistage-theme-label-light"
								},
								"tooltip": {
									"string_id": "onboarding-multistage-theme-tooltip-light-2"
								},
								"description": {
									"string_id": "onboarding-multistage-theme-description-light"
								}
							}, {
								"theme": "dark",
								"label": {
									"string_id": "onboarding-multistage-theme-label-dark"
								},
								"tooltip": {
									"string_id": "onboarding-multistage-theme-tooltip-dark-2"
								},
								"description": {
									"string_id": "onboarding-multistage-theme-description-dark"
								}
							}, {
								"theme": "alpenglow",
								"label": {
									"string_id": "onboarding-multistage-theme-label-alpenglow"
								},
								"tooltip": {
									"string_id": "onboarding-multistage-theme-tooltip-alpenglow-2"
								},
								"description": {
									"string_id": "onboarding-multistage-theme-description-alpenglow"
								}
							}]
						},
						"primary_button": {
							"label": {
								"string_id": "onboarding-multistage-theme-primary-button-label"
							},
							"action": {
								"navigate": true
							}
						},
						"secondary_button": {
							"label": {
								"string_id": "onboarding-multistage-theme-secondary-button-label"
							},
							"action": {
								"theme": "automatic",
								"navigate": true
							}
						}
					}
				}]
			}
		}
	}, {
		"slug": "treatment-a",
		"ratio": 1,
		"feature": {
			"featureId": "aboutwelcome",
			"enabled": true,
			"value": {
				"id": "msw-latest"
			}
		}
	}, {
		"slug": "control",
		"ratio": 1,
		"feature": {
			"featureId": "aboutwelcome",
			"enabled": true,
			"value": null
		}
	}],
	"targeting": "channel in [\"release\"] && version|versionCompare('82.!') >= 0 && localeLanguageCode == \"en\" && ((isFirstStartup && !('trailhead.firstrun.didSeeAboutWelcome'|preferenceValue)) || 'multi-stage-aboutwelcome-set-default-release-82-83' in activeExperiments) && 'app.shield.optoutstudies.enabled'|preferenceValue",
	"startDate": null,
	"endDate": null,
	"proposedDuration": 7,
	"proposedEnrollment": 7,
	"referenceBranch": null
}
```

### CODE REVIEW

File: app/experimenter/experiments/constants/nimbus.py [hunk 1]
```diff
@@ -33,15 +33,18 @@ class NimbusTargetingConfig:
...
         "locale using the release channel."
     ),
     targeting=(
-        "localeLanguageCode == 'en' "
-        "&& (isFirstStartup || experiment.slug in activeExperiments)"
+        'localeLanguageCode == "en" '
+        "&& ((isFirstStartup "
+        "&& !('trailhead.firstrun.didSeeAboutWelcome'|preferenceValue)) "
+        "|| '{experiment.slug}' in activeExperiments) "
+        "&& 'app.shield.optoutstudies.enabled'|preferenceValue"
```
Comment (k88hudson) [inline app/experimenter/experiments/constants/nimbus.py:47]: i'm not sure what the requirements are for this/if this is generally needed

File: app/experimenter/experiments/constants/nimbus.py [hunk 2]
```diff
@@ -121,7 +123,7 @@ class EmailType(models.TextChoices):
 
     EMAIL_EXPERIMENT_END_SUBJECT = "Action required: Please turn off your Experiment"
 
-    TARGETING_VERSION = "version|versionCompare('{version}') >= .!"
+    TARGETING_VERSION = "version|versionCompare('{version}') >= 0"
```
Comment (k88hudson) [inline app/experimenter/experiments/constants/nimbus.py:126]: :+1:

**Review by k88hudson — APPROVED** [review]

File: app/experimenter/experiments/constants/nimbus.py (hunk 1 above)
Comment (punamdahiya) [reply app/experimenter/experiments/constants/nimbus.py:64]: its safe to asume these are needed for first run about:welcome experiments

Comment (punamdahiya) [reply app/experimenter/experiments/constants/nimbus.py:64]: quick question, is it possible to further narrow targeting to windows 10 and en-US only?

**Pull request merged**